### PR TITLE
Add if not exists to Create Table As 

### PR DESCRIPTION
**Pull request by dshunter107**
## Checklist

 - [ ] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

Sorry to have to create it a second time. It seemed I made the mistake of merging instead of rebasing, and as a result, it led me to learn a very long and painful lesson of how git works, but the lesson was learned. I am using git cherry-pick to rebuild just my commits. I will have some more time later today to ensure that I got everything, but I did want to put some progress back on at least. 

Again apologies for the growing pains.

### CODE REVIEW

**Comment by matriv** [comment]
ok to test

File: libs/sql-parser/src/main/java/io/crate/sql/tree/CreateTableAs.java [hunk 1]
```diff
@@ -56,7 +58,12 @@ public boolean equals(Object o) {
         }
         CreateTableAs<?> that = (CreateTableAs<?>) o;
         return name.equals(that.name) &&
-               query.equals(that.query);
+            query.equals(that.query) &&
+            ifNotExists == that.ifNotExists;
+    }
```
Comment (mfussenegger) [inline libs/sql-parser/src/main/java/io/crate/sql/tree/CreateTableAs.java:63]: Can you also add the ifNotExists property to the hashCode calculation further below to make sure equals/hashCode are consistent?

Comment (dshunter107) [reply libs/sql-parser/src/main/java/io/crate/sql/tree/CreateTableAs.java:63]: okay. I will look into this.

Comment (dshunter107) [reply libs/sql-parser/src/main/java/io/crate/sql/tree/CreateTableAs.java:63]: I believe I did it.

**Comment by matriv** [comment]
@dshunter107 Can I please ask you to sign our [CLA](https://crate.io/community/contribute/cla/)?

**Comment by matriv** [comment]
@dshunter107 Are you planning to continue work on this PR soon, or should we take over?
The integration test you've added expects an exception to be thrown, which it shouldn't, so the new flag added at parser level is not yet used.

**Comment by dshunter107** [comment]
> @dshunter107 Are you planning to continue work on this PR soon, or should we take over? The integration test you've added expects an exception to be thrown, which it shouldn't, so the new flag added at parser level is not yet used.

Sorry this is taking so long. I got a bit busy with my first programming internship. If you have to take over, I completely understand. I was having a hard time understanding why my tests were passing on my computer, but not on github. I think with your last message, I might see what I was doing wrong.

**Comment by dshunter107** [comment]
> @dshunter107 Are you planning to continue work on this PR soon, or should we take over? The integration test you've added expects an exception to be thrown, which it shouldn't, so the new flag added at parser level is not yet used.

and yes, I am definitely willing to continue.

**Comment by dshunter107** [comment]
> @dshunter107 Are you planning to continue work on this PR soon, or should we take over? The integration test you've added expects an exception to be thrown, which it shouldn't, so the new flag added at parser level is not yet used.

@matriv Can you explain this a bit more? If I don't have it assert that an exception, the test will fail. Are you saying that the way that I did my integration test is incorrect? Or are you saying that I need to change the underlying code so that it does not throw an exception?

**Comment by matriv** [comment]
> @matriv Can you explain this a bit more? If I don't have it assert that an exception will occur, the test will fail. Are you saying that the way that I did my integration test is incorrect? Or are you saying that I need to change the underlying code so that it does not throw an exception?

The point of `IF NOT EXISTS` is to not throw an exception if the table already exists.
So far you've made the changes in the parser but you need to make the changes during execution to implement this. (do not throw exception).

**Comment by dshunter107** [comment]
got it 

> > @matriv Can you explain this a bit more? If I don't have it assert that an exception will occur, the test will fail. Are you saying that the way that I did my integration test is incorrect? Or are you saying that I need to change the underlying code so that it does not throw an exception?
> 
> The point of `IF NOT EXISTS` is to not throw an exception if the table already exists. So far you've made the changes in the parser but you need to make the changes during execution to implement this. (do not throw exception).

got it. Thanks!

**Comment by dshunter107** [comment]
@matriv 
I believed I fixed the problem of the error being thrown.

**Comment by matriv** [comment]
Thx for your effort @dshunter107!
I took over the PR from here as there is some more work to be done. You can follow the work here:
https://github.com/crate/crate/pull/15892

**Comment by dshunter107** [comment]
Okay. Thanks.

**Pull request closed without merging**